### ISSUE TITLE
[Bug]: error window outputs

### ISSUE THREAD

**Issue by shiyu22** [towhee-io/towhee#2451]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

1. Run with window:

```python
from towhee.dc2 import pipe
from towhee.operator.nop import NOPNodeOperator

p = (pipe.input('x')
         .map('x', 'z', lambda x: [i+1 for i in range(x)])
         .flat_map('z', 'x', NOPNodeOperator())
         .window('x', 'y', 2, 1, lambda x: sum(x))
         .map('y', 'z', lambda x: x*10)
         .output('z'))

res = p(4).to_list()
```
It will return `[[40]]` but the correct res is `[[30], [50], [70], [40]]`.

2. Run with time window:
```python
p = (pipe.input('x')
         .flat_map('x', 'x', lambda x: [i+1 for i in range(x)])
         .map('x', 'ts', lambda x: x*1000)
         .time_window('x', 'y', 'ts', 2, 2, lambda x: sum(x))
         .map('y', 'x', lambda x: x*10)
         .output('x'))

res = p(4).to_list()
```
It will return `[[30], [50], [70], [40], [4]]` but the correct res is `[[30], [50], [70], [40]]`.

### Environment

```markdown
- Towhee version(e.g. v0.1.3 or 8b23a93):
- OS(Ubuntu or CentOS):
- CPU/Memory:
- GPU:
- Others:
```

**Comment by junjiejiangjjj** [comment]
`z` get an error col type  @shiyu22 
```python
from towhee.dc2 import pipe
from towhee.operator.nop import NOPNodeOperator

p = (pipe.input('x')
         .map('x', 'z', lambda x: [i+1 for i in range(x)])
         .flat_map('z', 'x', NOPNodeOperator())
         .window('x', 'y', 2, 1, lambda x: sum(x))
         .map('y', 'z', lambda x: x*10)
         .output('z'))

res = p(4).to_list()

**Comment by shiyu22** [comment]
> `z` get an error col type @shiyu22
> 
> ```python
> from towhee.dc2 import pipe
> from towhee.operator.nop import NOPNodeOperator
> 
> p = (pipe.input('x')
>          .map('x', 'z', lambda x: [i+1 for i in range(x)])
>          .flat_map('z', 'x', NOPNodeOperator())
>          .window('x', 'y', 2, 1, lambda x: sum(x))
>          .map('y', 'z', lambda x: x*10)
>          .output('z'))
> 
> res = p(4).to_list()
> ```

Got it, and fixed in #2453
